Reworking .Inners on Managers, base classes, constructors to simplify logic

martinsawicki opened the pull request:
Porting Java SDK commits:
- https://github.com/Azure/azure-sdk-for-java/commit/cbf22d823a0dbd46d7055e034cf26e152982041e
- https://github.com/Azure/azure-sdk-for-java/commit/2bed83c202809b8aaf1a34a97cdff77ab09af4c0
- https://github.com/Azure/azure-sdk-for-java/pull/1494/commits/428315c391479ac30fc033a073e293932a1811eb
- partial https://github.com/Azure/azure-sdk-for-java/commit/bfbeb04a1c6a905e16ae4bc34c5067b1187f0f92
- partial https://github.com/Azure/azure-sdk-for-java/commit/3484bf40eabeee02cb60e3c857e6fd02b26b6c2c 
  - so far affecting WebApps, Batch, CDN, Compute, DNS
- https://github.com/Azure/azure-sdk-for-java/commit/643dbd97b10087e09bccf3c69dc30f7e0b75032b
- https://github.com/Azure/azure-sdk-for-java/commit/d2d119ab9fadda37a2eac724e42bc88dcc920d7e
- https://github.com/Azure/azure-sdk-for-java/commit/b41fdf5e8519f8364faecb708e64dd8e0db5bd5e
- https://github.com/Azure/azure-sdk-for-java/commit/b8167b60cf3632086dfed587d17338090c81f4b9

Commit:
exposing .Inner on any *Manager
(bringing to parity with Java SDK commit https://github.com/Azure/azure-sdk-for-java/commit/cbf22d823a0dbd46d7055e034cf26e152982041e)

azurecla commented:
Hi __@martinsawicki__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

Commit:
pushing IHasInner into GroupableResource and IndendependentChildResource
(porting Java SDK commit https://github.com/Azure/azure-sdk-for-java/commit/2bed83c202809b8aaf1a34a97cdff77ab09af4c0)

Commit:
simplifying references to Inner within collection impls

Commit:
simplifying constructor and inner tracking logic in Web Apps impls

Commit:
simplifying constructor logic with Inner in Batch

Commit:
simplifying CDN related impl constructor logic using Inners

Commit:
simplifying Compute impl constructor logic using Inners

Commit:
simplifying DNS-related constructors with Inners from Manager

Commit:
Merge remote-tracking branch 'upstream/Fluent' into Fluent

Commit:
Merge remote-tracking branch 'upstream/Fluent' into Fluent

Commit:
Test delay fix.

Commit:
Merge pull request #1 from hovsepm/martin-Fluent

Test delay fix.
